lib/handler/redirect.js

```diff
@@ -119,27 +119,28 @@ function parseLocation (statusCode, headers) {
 }
 
 // https://tools.ietf.org/html/rfc7231#section-6.4.4
-function shouldRemoveHeader (header, removeContent) {
+function shouldRemoveHeader (header, removeContent, unknownOrigin) {
   return (
     (header.length === 4 && header.toString().toLowerCase() === 'host') ||
-    (removeContent && header.toString().toLowerCase().indexOf('content-') === 0)
+    (removeContent && header.toString().toLowerCase().indexOf('content-') === 0) ||
+    (header.length === 13 && header.toString().toLowerCase() === 'authorization' && unknownOrigin)
```

Comment:
```suggestion
    (unknownOrigin && header.length === 13 && header.toString().toLowerCase() === 'authorization')
```

checking `unknownOrigin` first is faster.